Mage.Sets/src/mage/cards/d/DonalHeraldOfWings.java

```diff
@@ -0,0 +1,145 @@
+package mage.cards.d;
+
+import mage.MageIdentifier;
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.SpellCastControllerTriggeredAbility;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.OneShotEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.filter.FilterSpell;
+import mage.filter.predicate.Predicates;
+import mage.filter.predicate.mageobject.MageObjectReferencePredicate;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.stack.Spell;
+import mage.game.stack.StackAbility;
+import mage.game.stack.StackObject;
+import mage.players.Player;
+import mage.target.targetpointer.FixedTarget;
+import mage.util.functions.StackObjectCopyApplier;
+import mage.watchers.Watcher;
+
+import java.util.UUID;
+
+/**
+ * @author Alex-Vasile
+ */
+public class DonalHeraldOfWings extends CardImpl {
+
+    private static final FilterSpell filterSpell = new FilterSpell("a nonlegendary creature spell with flying");
+    static {
+        filterSpell.add(Predicates.not(SuperType.LEGENDARY.getPredicate()));
+        filterSpell.add(CardType.CREATURE.getPredicate());
+    }
+
+    public DonalHeraldOfWings(UUID ownderId, CardSetInfo setInfo) {
+        super(ownderId, setInfo, new CardType[]{CardType.CREATURE}, "{2}{U}{U}");
+
+        this.addSuperType(SuperType.LEGENDARY);
+
+        this.addSubType(SubType.HUMAN);
+        this.addSubType(SubType.WIZARD);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(3);
+
+        // Whenever you cast a nonlegendary creature spell with flying, you may copy it,
+        // except the copy is a 1/1 Spirit in addition to its other types.
+        // Do this only once each turn. (The copy becomes a token.)
+        // TODO: This still triggers and asks if you wanna use it, even if you've used it once this turn.
```

Comment:
I'm not sure how to handle this properly. The working is "Do this only once each turn". So I take that to mean that the ability should keep triggering until the person chooses to activate it since it's optional. But I'm not sure how to implement that.

Reply:
take a look at [[Nykthos Paragon]], it's the only other card that does this